src/UnitfulRecipes.jl

```diff
@@ -163,9 +163,35 @@ function append_unit_if_needed!(attr, key, label::Nothing, u)
     attr[key] = UnitfulString(string(u), u)
 end
 function append_unit_if_needed!(attr, key, label::S, u) where {S <: AbstractString}
-    if label ≠ ""
-        attr[key] = UnitfulString(S(string(label, " (", u, ")")), u)
+    if !isempty(label)
+        attr[key] = UnitfulString(S(surround(label,u, get(attr,:surround,:round))), u)
     end
 end
 
+#=============================================
+Surround unit string with specified delimiters
+=============================================#
+surround(l,u,f::Nothing) = string(l,' ',u)
+surround(l,u,f::Function) = f(l,u)
+surround(l,u,f::AbstractString) = string(l,f,u)
+surround(l,u,f::NTuple{2,<:AbstractString}) = string(l,f[1],u,f[2])
+surround(l,u,f::NTuple{3,<:AbstractString}) = string(f[1],l,f[2],u,f[3])
+surround(l,u,f::Char) = string(l,' ',f,' ',u)
+surround(l,u,f::NTuple{2,Char}) = string(l,' ',f[1],u,f[2])
+surround(l,u,f::NTuple{3,Char}) = string(f[1],l,' ',f[2],u,f[3])
+surround(l,u,f::Bool) = f ? surround(l,u,:round) : surround(l,u,nothing)
+function surround(l,u,f::Symbol)
+    f===:round && return surround(l,u,('(',')'))
+    f===:square && return surround(l,u,('[',']'))
+    f===:curly && return surround(l,u,('{','}'))
+    f===:angle && return surround(l,u,('<','>'))
+    f===:slash && return surround(l,u,'/')
+    f===:slashround && return surround(l,u,(" / (",")"))
+    f===:slashsquare && return surround(l,u,(" / [","]"))
+    f===:slashcurly && return surround(l,u,(" / {","}"))
+    f===:slashangle && return surround(l,u,(" / <",">"))
+    f===:verbose && return surround(l,u," in units of ")
+end
```

Comment:
Interesting, I think I would have done this with a constant `Dict` and then do something like
```julia
surround(l, u, f::Symbol) = surround(l, u, SURROUND_DICT[f])
```
Is this function better than a `Dict` for some reason? (Not asking to change this, just curious)

Reply:
I don't know. A dict has a different memory layout, but is probably indexed more efficiently (this is essentially a linear search). But like this we can order things by how likely they are to be used, so every time someone uses `:round` (which is the default) it's just a single comparison.

I don't think there's a measurable difference in performance for this case.

Reply:
FWIW I would also have used a `Dict` for such a switch that is not a performance bottleneck. Why? Because a `Dict` is easier to inspect (e.g. `display` gives a ton of information) than a function. 
Anyway it is not a huge deal and you can leave it as is.